### PR TITLE
[DPP-1294] Cleaning up AcsReader

### PR DESCRIPTION
**Pull request by pbatko-da**
changelog_begin
changelog_end

Renamings, expanding the commentary, removing ACSReader trait.
No changes in functionality.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala [hunk 1]
```diff
@@ -19,86 +19,89 @@ import scala.annotation.tailrec
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.chaining._
 
-trait ACSReader {
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala:22]: Removing this trait

File: ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceOwner.scala [hunk 2]
```diff
@@ -167,12 +167,12 @@ final class IndexServiceOwner(
       dbSupport = dbSupport,
       eventsPageSize = config.eventsPageSize,
       eventsProcessingParallelism = config.eventsProcessingParallelism,
-      acsIdPageSize = config.acsIdPageSize,
-      acsIdPageBufferSize = config.acsIdPageBufferSize,
-      acsIdPageWorkingMemoryBytes = config.acsIdPageWorkingMemoryBytes,
-      acsIdFetchingParallelism = config.acsIdFetchingParallelism,
-      acsContractFetchingParallelism = config.acsContractFetchingParallelism,
-      acsGlobalParallelism = config.acsGlobalParallelism,
+      acsMaxNumberOfIdsPerIdPage = config.acsIdPageSize,
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceOwner.scala:170]: Does this rename bring value, given that now the chain of names is broken and the same name cannot be followed from hocon all the way to the implementation?

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala [hunk 3]
```diff
@@ -19,86 +19,89 @@ import scala.annotation.tailrec
...
-class FilterTableACSReader(
-    dispatcher: DbDispatcher,
+/** Streams ACS events (active contracts) in a two step process consisting of:
+  * 1) fetching event sequential ids of the active contracts based on the filtering constraints,
+  * 2) fetching the active contracts based on the fetched event sequential ids.
+  *
+  * Details:
+  * An input filtering constraint (consisting of parties and template ids) is decomposed into
+  * a number of simple filtering constraints (a constraint with one party and at most one template id).
+  * For each simple filter in parallel the matching event sequential ids are fetched and then merged into
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala:29]:
```suggestion
  * For each simple filter, the matching event sequential ids are fetched in parallel and then merged into
```

File: ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceOwner.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceOwner.scala:170]: I see two issues:
1) being able to independently manage config keys names vs. corresponding names used in the Scala code,
2) having well and consistently named config keys and Scala names.




FYI I'm putting this PR on hold

**Pull request closed without merging**